File: server/model/store.go
package model

type Store struct {
	ID              int64  `json:"id"`
	Number          string `json:"number" validate:"required"`
	Name            string `json:"name" validate:"required"`
	CorporateName   string `json:"corporate_name"`
	Address         string `json:"address" validate:"required"`
	City            string `json:"city" validate:"required"`
	State           string `json:"state" validate:"required,len=2"`
	ZipCode         string `json:"zip_code" validate:"required"`
	AddressNumber   string `json:"address_number" validate:"required"`
	EstablishmentID int64  `json:"establishment_id" validate:"required"`
}

File: web/src/types/store.ts
export interface Store {
  number: string;
  name: string;
  corporate_name: string;
  address: string;
  address_number: string;
  city: string;
  state: string;
  zip_code: string;
}

export interface StoreWithEstablishment extends Store {
  id: string;
  businessId: string;
  establishmentId: string;
}